File: Array & Hashing/1. Two Sum/TwoSum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            Integer j = map.get(nums[i]);
            if(j!=null) return new int[] {j, i};
            map.put(target-nums[i], i);
        }
        return null;
    }
}

File: Array & Hashing/1. Two Sum/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for i,num in enumerate(nums):
            if num in d:
                return [i, d[num]]
            d[target-num] = i